frontend/openchat-client/src/openchat.ts

```diff
@@ -4404,16 +4403,15 @@ export class OpenChat extends OpenChatAgentWorker {
                     await updateRegistryTask;
                 }
 
-                const updatedChats = (chatsResponse.state.directChats as ChatSummary[])
+                const chats = (chatsResponse.state.directChats as ChatSummary[])
                     .concat(chatsResponse.state.groupChats)
                     .concat(chatsResponse.state.communities.flatMap((c) => c.channels));
 
-                this.updateReadUpToStore(updatedChats);
-                const chats = getAllServerChats(this._liveState.globalState).values();
+                this.updateReadUpToStore(chats);
 
-                this._cachePrimer?.processChatUpdates(chats, updatedChats);
+                this._cachePrimer?.processChats(chats);
```

Comment:
Does this mean, now that we are not using the global store to `getAllServerChats` that we could potentially move the cache primer entirely into the worker? That has always seemed like where it belongs. And would also be better if and when we can convert the worker to a service worker (so everything could be kept up to date offline).

Reply:
Ooooo yeah that sounds good let me try that.